main.py

```diff
@@ -102,12 +103,13 @@ def get_inps(
 
     dtype = next(iter(model.parameters())).dtype
     nsamples_per_device = (len(data) - 1) // len(devices) + 1
+
     inps = [
         torch.zeros(
             (min(nsamples_per_device, len(data) - i * nsamples_per_device), model_seqlen, model.config.hidden_size),
             dtype=dtype,
             device=devices[i] if not offload_activations else "cpu",
-            pin_memory=offload_activations,
+            pin_memory=offload_activations if torch.cuda.is_available() else False,
```

Comment:
1) I think that the better way to check if we use cpu training will be to check if `devices == [torch.device('cpu')]`. This way user with gpus will be able to use cpu training with `--devices cpu` flag.
2) Let's use `A and B` instead of `A if B else False` statement (if you don't have any good reason to do otherwise).
3) I see a lot of code regarding offload_activations. Does cpu training work without this flag? Looks like it should.